File: projetoES3/src/model/domain/Estado.java
package model.domain;

public class Estado {

	private String siglaEstado;
	private String descricaoEstado;
	private Pais paisEstado;

	public Estado(String sigla, String descricao, Pais pais) {
		this.siglaEstado = sigla;
		this.descricaoEstado = descricao;
		this.paisEstado = pais;
	}
	
	
	public Estado() {	
	}
	
	public Estado(String descricao) {	
		this.descricaoEstado = descricao;
	}
	
	public Estado(String descricao, Pais pais) {	
		this.descricaoEstado = descricao;
		this.paisEstado = pais;
	}

	public String getSigla() {
		return siglaEstado;
	}

	public void setSigla(String sigla) {
		this.siglaEstado = sigla;
	}

	public String getDescricao() {
		return descricaoEstado;
	}

	public void setDescricao(String descricao) {
		this.descricaoEstado = descricao;
	}
	
	public Pais getPais() {
		return paisEstado;
	}

	public void setPais(Pais pais) {
		this.paisEstado = pais;
	}
}

File: typeScript/src/model/domain/Estado.ts
import {Pais} from "./Pais";

export class Estado {
    constructor(
        private id: number,
        private  siglaEstado: string,
        private  descricaoEstado: string,
        private  paisEstado: Pais
    ) {
    }

    /*public Estado(descricao: string): void {
        this.descricaoEstado = descricao;
    }*/

    public estado(descricao: string, pais: Pais): void {
        this.descricaoEstado = descricao;
        this.paisEstado = pais;
    }

    public getId(): number {
        return this.id;
    }

    public setId(sigla: number): void {
        this.id = sigla;
    }

    public getSigla(): string {
        return this.siglaEstado;
    }

    public setSigla(sigla: string): void {
        this.siglaEstado = sigla;
    }

    public getDescricao(): string {
        return this.descricaoEstado;
    }

    public setDescricao(descricao: string): void {
        this.descricaoEstado = descricao;
    }

    public getPais(): Pais {
        return this.paisEstado;
    }

    public setPais(pais: Pais): void {
        this.paisEstado = pais;
    }
}


